File: 0520-detect-capital/0520-detect-capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        int count = 0;

        for(int i = 0; i < word.length(); i++) {
            if((int)word.charAt(i) >= 65 && (int)word.charAt(i) <= 90)
            count++;
        }

        if(((int)word.charAt(0) >= 65 && (int)word.charAt(0) <= 90) && count == 1 || count == word.length() || count == 0)
            return true;

        return false;
    }
}

File: 0520-detect-capital/0520-detect-capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        if len(word) == 1 :
            return True

        cap_count = 0
        count = 0

        for i in range (len(word)) :
            if ('A' <= word[i] <= 'Z') :
                cap_count += 1
            else :
                count += 1

        if cap_count == len(word) :
            return True

        if count == len(word) :
            return True

        if ('A' <= word[0] <= 'Z' and cap_count == 1 and count == len(word) - 1) :
            return True

        return False